### ISSUE TITLE
Ability to run the install script without having to answer any questions

### ISSUE THREAD

**Issue by Romeo-Golf** [jprochazka/adsb-receiver#251]
As the install script is used to update both itself, and the various component software elements, it would seem useful to have an option to run in such a way that it can be left unattended and possibly even regularly via cron..

Examples of this are Debian's apt-get which can be used with '-y' option to say yes to any questions; however the adsb-exchange setup does have some questions where the default answer is not yes so perhaps we could have:

-y     yes to all questions, basically a full (re)install
-n     no to all questions, perhaps could check for updates but not install anything
-d    default answer to all questions,

If my understanding is correct then accepting the default answer to all questions, either with this option of just by hitting enter at every prompt, should result in a basic install when run for the first time and would update/reinstall all currently installed packages when run on an existing install... is this the case?

**Comment by jprochazka** [comment]
Two things which may pertain to this request.

I will be adding a configuration file into the mix shortly. Last week it was asked if WiFi could be configured much how FlightAware does it on their image. When looking into adding this as a feature I will look at also implementing a way to specify what is to be installed. This would cover the images...

The same could be done on the script side making it a headless installation process as well. Maybe a configuration file in the root of the project directory containing a list of the options to install.

I will defiantly look into adding this I would first want to complete Issue #50 so that if any issues are encountered they can be hunted down.

**Comment by Romeo-Golf** [comment]
Would it be viable to add the ability to parse commands/flags when running the install.sh script?

If so then it should be trivial to make the 'press enter to continue..' dialogs boxes only displayed when a verbose flag is passed:

>    if [ ${VERBOSE} ] ; then
>        read -p "Press enter to continue..." CONTINUE
>    fi

And perhaps even have an EXPERT or REINSTALL flag that is able to answer some of the questions that are more complex than press enter to continue ;)

**Comment by jprochazka** [comment]
I plan to make this fairly hands free. The only stoppages would be if an error is encountered or at the end. This will also tie into the logging to text file feature to be added as well so that the user will have the ability to review what was done after the fact.

**Comment by Romeo-Golf** [comment]
So I had a go at making the "Press enter to continue..." boxes only appear if install.sh is called with the -v or --verbose options, as this is a fairly quick and dirty change I submitted the pull request against 2.5.1 but please let me know if you would rather I submitted against 2.6.0 as this feature request is targeted for ;)

**Comment by jprochazka** [comment]
Actually submit it to the 2.6.0 branch. That is where I am currently working on issues #50, #262 and #251.

The way things work with the branches are as follows:

The x.x.1 branch is there for bug fixes.
Generally these are merged into the master branch from this branch right away since the addition of the new upgrade system. If it is a game breaking bug then the a tag would be created and a release made named x.x.1. If this were to occur then a new x.x.2 branch would be made which would be used the same as x.x.1 was before it became a release.

A 1.x.x or x.1.x branch would be for upcoming major and minor releases. In the case of the 2.6.0 branch which at this time is the current "next release" branch any items listed in Milestone v2.6.0.

I will add a contribute section to the wiki when I get the free time explaining this better,

**Comment by Romeo-Golf** [comment]
Resubmitted against 2.6.0 with slightly tidier code, -h or --help shows the usage and -v or --verbose is now required to see the "Press enter to continue..." prompts; however regardless of the option parse it will set the VERBOSE flag if it hits the existing code which suggests it to be a new install :)

Is it worth me doing something similar to the whiptails whereby another option such as --expert or --upgrade which will be equivalent to all of the default answers to the whiptail dialog boxes, thus potentially allowing for the script to be regularly called by cron to update at least the OS and its own code, if not the decoders and feeders?

**Comment by Romeo-Golf** [comment]
Also as an automated re-install, potentially called from cron, could potentially result in a brief interuption to the feeder I thought it could be worth introducing a pseudo-random delay of between 5 and 59 minutes:

function RandomDelay() {
 if [ ${DELAY} ] ; then
  DELAY_TIME=`echo "(( 300 + ( $RANDOM + $RANDOM )) / 20 )" | bc`
  DATE=`date`
  echo -e ""
   echo -e "\e[91m $ADSB_PROJECTTITLE"
   echo -e ""
   echo -en "\e[92m  Pausing for ${DELAY_TIME} seconds from: \t\e[39m $DATE"
   echo -e ""
   sleep $DELAY_TIME
  fi
}

RandomDelay


Is it worth me messing about with pull requests or would you be happy to factor this in at a suitable point in your ongoing rewrites and improvements of the codebase? :)

**Comment by Romeo-Golf** [comment]
I have merged my changes back in with your changes to move code from install.sh to bash/init.sh as pull #275

**Comment by Romeo-Golf** [comment]
I see the Verbose flag is being replaced by the Automated install; however if possible it would be good to retain this functionality for expert/developer install.... 

Eg while testing my changes I probably ran install.sh at least a hundred times, if I'd had to hit enter at each of the prompts I'd probably have got bored and given up...

So while it will be great when the automated install option completed, I still see a use case for a streamlined (re)install where the user is prompted for any required information but not unduely slowed down by having to hit enter a dozen times ;)

Do you see any issues with retaining this functionality, either with the press enter to continue prompts only being shown when the variable is set (as the VERBOSE flag currently does) or the inverse whereby the flag must be set in order for these prompts to be shown?
 with whatever term you feel appropriate for either only prompting to hit enter to continue if

**Comment by jprochazka** [comment]
There will be an additional flag like yours. When merged I didn't notice at first that by default all the stops would be skipped. I need/want these stops to be hit by default so that people are able to review what exactly happened during installation if an error is encountered. This helps greatly when having to support issues a user may encounter.

Following your example another switches will be added such as -f --fast or something to that nature which a user can add to skip these stops if they so desire. I have been meaning to contact you regarding this after the holiday which has caused a slight stoppage to development.

**Comment by Romeo-Golf** [comment]
yeah no worries, more than happy with the the inverse of the verbose flag as either -f|--fast, -s|--skip-stops or perhaps even -q|--quick ?

will drop you an email once ive finished tidying up the RTLSDR-OGN code to avoid holding up 2.6.0 ;)
